### PR TITLE
 Do not break when sample has not existing reference

### PR DESCRIPTION
**Pull request by lslachciak**
[Issue] N/A
[Problem] Some downloaded samples have missing references
[Solution] Ignore error when sample is referencing to not existing file.
Only stop when sample does not exist at all

Signed-off-by: Lukasz Slachciak <l.slachciak@samsung.com>

### CODE REVIEW

File: routes/demos.js [hunk 1]
```diff
@@ -55,12 +55,19 @@ module.exports = function (express) {
...
+          exec(`wget --spider ${sampleUrl}`, (error, stdout, stderr) => {
             if (error) {
               console.log(error);
               return callback(`Could not download sample from ${sampleUrl}`);
             }
-            callback(null, project, projectPath);
+            else {
+              exec(`wget --page-requisites --convert-links --no-host-directories --cut-dirs=${numDirsToCut} --directory-prefix ${projectPath} ${sampleUrl}`, (error, stdout, stderr) => {
+                if (error) {
+                  console.log(error);
```
Comment (grzegorz-czajkowski) [inline routes/demos.js]: Please add a comment why we don't return.

File: routes/demos.js [hunk 2]
```diff
@@ -55,12 +55,19 @@ module.exports = function (express) {
           const sampleUrl = new URL(samplePath, tauExamplesHost);
           // Define number of unnecessary directories to be omitted while downloading.
           const numDirsToCut = sampleUrl.pathname.match(/TAU\/examples\/mobile|wearable\/UIComponents/g) ? 4 : 0;
```
Comment (grzegorz-czajkowski) [inline routes/demos.js]: numDirsToCut is not needed here. Please move it below.

File: routes/demos.js [hunk 3]
```diff
@@ -55,12 +55,19 @@ module.exports = function (express) {
...
-          exec(`wget --page-requisites --convert-links --no-host-directories --cut-dirs=${numDirsToCut} --directory-prefix ${projectPath} ${sampleUrl}`, (error, stdout, stderr) => {
+          exec(`wget --spider ${sampleUrl}`, (error, stdout, stderr) => {
```
Comment (grzegorz-czajkowski) [inline routes/demos.js]: How about adding comment for this line?

File: routes/demos.js (hunk 1 above)
Comment (lslachciak) [reply routes/demos.js]: done

File: routes/demos.js (hunk 2 above)
Comment (lslachciak) [reply routes/demos.js]: done

File: routes/demos.js (hunk 3 above)
Comment (lslachciak) [reply routes/demos.js]: done

**Comment by lslachciak** [comment]
Commit message  fixed + @grzegorz-czajkowski  remarks - done

**Pull request merged**